api/response/schema_query.py

```diff
@@ -102,49 +113,24 @@ def resolve_room_response(self, info, room_id):
             total_room_resources=total_room_resources,
             response=responses)
 
-    def get_all_reponses(self, info):
-        response = []
-        rooms = RoomModel.query.all()
-        for room in rooms:
-            room_name = room.name
-            room_response = Response.get_query(info).filter_by(room_id=room.id)
-            total_response = room_response.count()
-            all_responses, total_room_resources = Query.get_room_response(
-                self, room_response, room.id)
-            responses = RoomResponse(
-                room_id=room.id,
-                room_name=room_name,
-                total_responses=total_response,
-                total_room_resources=total_room_resources,
-                response=all_responses)
-            response.append(responses)
-        return response
-
-    def filter_rooms_by_responses(self, info, upper_limit, lower_limit):
-        all_responses = Query.get_all_reponses(self, info)
-        filtered_responses = []
-        for response in all_responses:
-            reponse_count = response.total_responses
-            if lower_limit <= reponse_count <= upper_limit:
-                filtered_responses.append(response)
-        return filtered_responses
-
-    def search_response_by_room(
-        self, info, upper_limit, lower_limit, room
-    ):
+    def search_response_by_room(self, info, upper_limit, lower_limit, room):
         search_result = []
         filtered_search = []
-        if upper_limit and lower_limit:
-            filtered_response = Query.filter_rooms_by_responses(
-                self, info, upper_limit, lower_limit)
-            for response in filtered_response:
-                if response.room_name.lower() == room.lower():
-                    filtered_search.append(response)
-            if filtered_search:
-                return filtered_search
-            else:
-                raise GraphQLError(
-                    "No response for this room at this range")
+        if isinstance(upper_limit, int):
+            filtered_count_search = validate_responses_by_room(
+                    int, filter_rooms_by_responses, upper_limit=upper_limit,
+                    lower_limit=lower_limit, room=room, Query=Query, info=info,
+                    filtered_search=filtered_search
+                )
+            if filtered_count_search:
```

Comment:
What would happen if you just return `filtered_count_search` without the `if statement`?

Reply:
If a user inputs an invalid room, then the response would return null instead of throwing the expected error `No response for this room, enter a valid room name`. as seen below
<img width="897" alt="Screen Shot 2019-04-09 at 1 29 40 PM" src="https://user-images.githubusercontent.com/22786444/55793693-33c6b700-5acc-11e9-9005-265553900e68.png">
**expected response**
<img width="1147" alt="Screen Shot 2019-04-09 at 1 29 56 PM" src="https://user-images.githubusercontent.com/22786444/55793739-43460000-5acc-11e9-8fbe-5d2d0374d1f4.png">